### PR TITLE
add scroll overshoot bounce animation

### PR DESCRIPTION
**Pull request by Frizi**
### Pull Request Description

Implements https://www.pivotaltracker.com/n/projects/2539304/stories/183390749

https://user-images.githubusercontent.com/919491/198984088-9c4f03db-922f-4503-83fa-f0301a11239e.mp4

### Important Notes

- The overshoot bounce animation is implemented as a animation-like FRP mixin. It is currently defined using `define_endpoints` macro, but should be migrated to `define_endpoints_2` as soon as it supports "frp mixins" - definitions that extend external networks.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: lib/rust/ensogl/component/scrollbar/src/lib.rs [hunk 1]
```diff
@@ -57,32 +57,37 @@ const HIDE_DELAY: f32 = 1000.0;
 const CLICK_AND_HOLD_DELAY_MS: i32 = 500;
 /// Time interval between scrolls while holding down a mouse button, in milliseconds.
 const CLICK_AND_HOLD_INTERVAL_MS: i32 = 200;
-
+/// Maximum scroll overshoot in pixels.
+const SCROLL_OVERSHOOT_LIMIT: f32 = 60.0;
```
Comment (MichaelMauderer) [inline lib/rust/ensogl/component/scrollbar/src/lib.rs]: This should be part of the theme, to allow easier configuration.

File: lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    /// Public FRP api.
+    pub frp: FrpEndpoints,
+}
+
+impl OvershootAnimation {
+    /// Constructor. Hooks into preexisting network. Created `OvershootAnimation` struct does not
+    /// need to be persisted, all created FRP nodes will be managed by the passed-in network.
+    pub fn new(network: &frp::Network) -> Self {
+        let frp = Frp::extend(network);
+        // let out = &frp.private().output; // version for `define_endpoints_2`
```
Comment (MichaelMauderer) [inline lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs]: If this stays in the code, it should be annotated with a proper TODO pointing to a pivotal issue where it is resolved.

**Review by MichaelMauderer — CHANGES_REQUESTED** [review]

File: lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+// ==================
+// === Constants  ===
+// ==================
+
+/// Determines how much of out-of-bounds portion of value change is actually applied to the
+/// animation target value. Larger values will make the overshoot more noticeable.
+const OUT_OF_BOUNDS_SENSITIVITY: f32 = 0.3;
+
+
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs:29]: spacing - 3 newlines before section

File: lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+// ==========================
+// === OvershootAnimation ===
+// ==========================
+
+/// An [`Animation`] wrapper that allows the value to be smoothly manipulated within set bounds,
+/// overshoot set bounds and bounce back.
+#[derive(Clone, CloneRef, Debug, Shrinkwrap)]
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs]: This does not explain how to use this structure and how it works. I feel that this is a very cool struct, it should have really good docs.

**Review by wdanilo — APPROVED** [review]

File: lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs (hunk 4 above)
Comment (Frizi) [reply lib/rust/ensogl/core/src/animation/frp/animation/overshoot.rs]: Added more detailed documentation how to use the provided API and what's the animation behavior. There is also an overview already in the top of the module.

**Review by MichaelMauderer — APPROVED** [review]

**Pull request merged**